src/main/java/com/tysng/duke/service/Duke.java

```diff
@@ -0,0 +1,103 @@
+package com.tysng.duke.service;
+
+import com.tysng.duke.domain.Task;
+import com.tysng.duke.exception.CommandException;
+import com.tysng.duke.storage.Storage;
+import com.tysng.duke.ui.Command;
+import com.tysng.duke.ui.Response;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This class handles the logic of the Application.
+ *
+ * <p>For every Command, it will process and return a Response object. This class
+ * also interacts with te Storage class to retrieve and save TaskList objects to local storage.
+ */
+public class Duke {
+    private TaskList taskList;
+    private Storage storage;
+
+    /**
+     * Constructs a service layer object with the external Storage layer.
+     *
+     * @param storage an instance of the Storage object
+     */
+    public Duke(Storage storage) {
+        this.storage = storage;
+        this.taskList = new TaskList(this.storage.loadTasks());
+    }
+
+    /**
+     * Processes a Command object. This is the driver method of the class.
+     *
+     * @param command a successfully parsed user command
+     * @return the corresponding response for the given command
+     */
+    public Response handle(Command command) {
```

Comment:
It might be cleaner to subclass `Command` and implement an `execute` method for each subclass instead of using enum and switch IMO.